lib/mayaUsd/ufe/UsdHierarchy.cpp

```diff
@@ -270,12 +309,20 @@ Ufe::AppendedChild UsdHierarchy::appendChild(const Ufe::SceneItem::Ptr& child)
 Ufe::InsertChildCommand::Ptr
 UsdHierarchy::insertChildCmd(const Ufe::SceneItem::Ptr& child, const Ufe::SceneItem::Ptr& pos)
 {
+    // Changing the hierarchy of inactive items is not allowed.
+    if (!fItem->prim().IsActive())
+        return nullptr;
+
     return UsdUndoInsertChildCommand::create(fItem, downcast(child), downcast(pos));
 }
 
 Ufe::SceneItem::Ptr
 UsdHierarchy::insertChild(const Ufe::SceneItem::Ptr& child, const Ufe::SceneItem::Ptr& pos)
 {
+    // Changing the hierarchy of inactive items is not allowed.
+    if (!fItem->prim().IsValid())
+        return nullptr;
+
     auto insertChildCommand = insertChildCmd(child, pos);
     return insertChildCommand->insertedChild();
```

Comment:
You're repeating the test, and now insertChildCmd() can return a nullptr, which you should check.  Yes, currently, since the test is the same, you're avoiding nullptr deref, but because there are two tests, they could get out of sync.  I would just remove the duplicate test here.
```suggestion
    auto insertChildCommand = insertChildCmd(child, pos);
    return insertChildCommand ? insertChildCommand->insertedChild() : nullptr;
```

Reply:
Oh I failed to notice they called each other, I did not read the implementation, I just put the check in every function that was related to hierarchy changes. I'll try your changes.